db/sqlinterfaces.c

```diff
@@ -4539,15 +4539,21 @@ static int execute_verify_indexes(struct sqlthdstate *thd,
         clnt->has_sqliterow = 1;
         rc = verify_indexes_column_value(stmt, clnt->schema_mems);
         if (gbl_enable_internal_sql_stmt_caching) {
-            stmt_cache_add_entry(thd->stmt_cache, clnt->sql, 0, stmt, clnt);
```

Comment:
I think we should just finalize/free this stmt here if `stmt_cache_add_entry` returns `-1`, as `stmt_cached_add_entry` already checks for duplicates.

Reply:
We discussed this; the cached stmt was also removed during find() so stmt_cache_add_entry wouldn't fail.